src/util/ringdelaybuffer.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+* $Id$
+* Portable Audio I/O Library
+* Ring Buffer utility.
+*
+* Author: Phil Burk, http://www.softsynth.com
+* modified for SMP safety on OS X by Bjorn Roche.
+* also allowed for const where possible.
+* modified for multiple-byte-sized data elements by Sven Fischer
+*
+* This program is distributed with the PortAudio Portable Audio Library.
+* For more information see: http://www.portaudio.com
+* Copyright (c) 1999-2000 Ross Bencina and Phil Burk
+*
+* Permission is hereby granted, free of charge, to any person obtaining
+* a copy of this software and associated documentation files
+* (the "Software"), to deal in the Software without restriction,
+* including without limitation the rights to use, copy, modify, merge,
+* publish, distribute, sublicense, and/or sell copies of the Software,
+* and to permit persons to whom the Software is furnished to do so,
+* subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be
+* included in all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR
+* ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+* CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+* WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+/*
+* The text above constitutes the entire PortAudio license; however,
+* the PortAudio community also makes the following non-binding requests:
+*
+* Any person wishing to distribute modifications to the Software is
+* requested to send the modifications to the original developer so that
+* they can be incorporated into the canonical version. It is also
+* requested that these non-binding requests be included along with the
+* license above.
+*/
+
+#include "ringdelaybuffer.h"
+
+#include "util/math.h"
+#include "util/sample.h"
+
+using ReadableSlice = mixxx::SampleBuffer::ReadableSlice;
+using WritableSlice = mixxx::SampleBuffer::WritableSlice;
+
+namespace {
+SINT copyRing(const ReadableSlice pSourceBuffer,
+        SINT sourcePos,
+        const WritableSlice pDestBuffer,
+        SINT destPos,
+        const SINT numItems) {
+    const SINT newSourcePos = sourcePos + numItems;
+    const SINT newDestPos = destPos + numItems;
+
+    VERIFY_OR_DEBUG_ASSERT(newSourcePos <= pSourceBuffer.length() ||
+            newDestPos <= pDestBuffer.length()) {
+        return 0;
+    }
```

Comment:
Than we have half the destied result which may worse. Since this is a not used case in this anonymous namespace function, asset and return 0 is a good choice. IMHO